lib/components/admin/field-trip-list.js

```diff
@@ -127,6 +159,32 @@ class FieldTripList extends Component {
   }
 }
 
+class Tab extends Component {
+  getCount = memoize((data, filter) => data.filter(filter).length)
+
+  render () {
+    const {data, filter, onClick, tab} = this.props
+    const active = tab.id === filter.tab
+    const style = {
+      backgroundColor: active ? 'navy' : undefined,
+      borderRadius: 5,
+      color: active ? 'white' : undefined,
+      padding: '2px 3px'
+    }
+    return (
+      <button
+        className='clear-button-formatting'
+        key={tab.id}
+        name={tab.id}
+        onClick={onClick}
+        style={style}
+      >
+        {tab.label} <Badge>{this.getCount(data, tab.filter)}</Badge>
+      </button>
+    )
+  }
+}
```

Comment:
I guess using `lodash.memoize` works here. However, since `lodash.memoize` caches based on the first argument, it's accomplishing the memorization without the second argument affecting the cache key calculation, which is fine since that argument wouldn't get updated.  Another way of doing this without lodash.memoize would be to use React's `useMemo`:

```suggestion
function Tab ({data, filter, onClick, tab}) {
  const count = React.useMemo(() => data.filter(tab.filter).length, [data, tab])
  const active = tab.id === filter.tab
  const style = {
    backgroundColor: active ? 'navy' : undefined,
    borderRadius: 5,
    color: active ? 'white' : undefined,
    padding: '2px 3px'
  }
  return (
    <button
      className='clear-button-formatting'
      key={tab.id}
      name={tab.id}
      onClick={onClick}
      style={style}
    >
      {tab.label} <Badge>{count}</Badge>
    </button>
  )
}
```

However, in either of the `lodash.memoize` or `useMemo` approaches, if the FieldTripList component is unmounted then these Tab instances and the cache saved in those `lodash.memoize`/`useMemo` instances are lost. I'm not sure how often call-takers would close and then reopen the window and how much of a performance hit this would be to recalculate everything. If that were a big deal, then another way to do this would be to use reselect to compute the values for all of the tab counts and return an object where the count values could be looked up by `tab.id`.

Reply:
The main perf case I was trying to resolve was when users search or switch tabs. This looks great, thanks for sharing the suggestion.